code/android/src/main/java/no/bakkenbaeck/porchpirateprotector/manager/SharedPreferencesManager.kt

```diff
@@ -67,7 +66,7 @@ class SharedPreferencesManager(
 
     override fun removeIPAddress(address: String) {
         loadIPAddresses()?.let { initialAddresses ->
-            val updated = initialAddresses.removeIfPresent(address)
+            val updated = initialAddresses.filter { it == address }
```

Comment:
Personally I think if you can refactor `removeIfPresent` to use this under the hood, that'd be better, since it makes the intent of the method much more explicit. 

What do you feel is the advantage of just swapping this to `filter`?

Reply:
`filter` is a standard Kotlin operation -- since it's a standard feature of the library its purpose should be assumed to be known.

Having a `filter` operation inside a function called `removeIPAddress` should be all the implied intention you need.

`removeIfPresent` is just a layer of abstraction for the sake of it -- it's just a semantic layer on top of standard Kotlin. `removeIfPresent` is also a poor name if you were to rewrite it to using the `filter` operation, because the `IfPresent` part of the method name is implicit, and also isn't that important to how `filter` works. So realistically you'd have to rename the extension function to `remove` which just brings home how meaningless it is to have this layer of abstraction.

Not to mention, theres really no need for this to be an extension function.

Reply:
I mean, yes, it's a standard expression function, but `filter` is flexible enough to do a bazillion things  - having an alias that makes it clearer what you're doing *can* be helpful - I'm not saying a `remove` that just calls through to `filter` wouldn't  be helpful from a standpoint of readability.

Having said that, on balance I think you're correct that `removeIPAddress` makes it clear enough what this is doing that a `remove` function is not a necessary clarification, so 👍.